### ISSUE TITLE
Update missions with gadgetry

### ISSUE THREAD

**Issue by sprunk** [ZeroK-RTS/Zero-K#3793]
91bc8f0a696bd826c236c2b3f6d3009c386be435

Similar to #3784, but for missions and the custom keys tables from `GetPlayerInfo` and `GetTeamInfo`.

**Comment by sprunk** [comment]
Some time ago I tried one of the Sunrise missions and there was some issue running it. Perhaps they would require to be fully ported anyway. In that case the task would be to make sure the Mission Editor doesn't put bad code into new missions and to check the handful of ME missions in the campaign (folsom + koda rally).

**Comment by GoogleFrog** [comment]
I hear BAR is working on a scripted campaign with gadgets. Take their framework when it is done.

**Comment by sprunk** [comment]
Yes, I more or less plan to do that. But they don't seem to be progressing at a meaningful rate. Meanwhile I was recently surprised that somebody was using ZKME with reasonable success.

**Comment by GoogleFrog** [comment]
Hm ok. Imo the missions should never have contained gadgets, they should have been maintained in the game. Perhaps try to do that? Apparently Kodachi Rally works.

**Comment by sprunk** [comment]
Yes. IMO ideally gadgets that have some level of generality (which includes everything general enough to be exposed in ZKME) should be split into a separate gadget strictly exposing an API (but not using it) and a separate gadget implementing the ZK defaults using that API. As a specific example: #5043

This ticket was just about making sure missions don't have gadgets specifically using obsolete `Spring.GetTeamInfo` and `Spring.GetPlayerInfo` calls though.
